File: Baekjoon/Swift/BOJ_PS/BOJ_PS/Stack-Queue/Q1874.swift
//
//  Q1874.swift
//  BOJ_PS
//
//  Created by 김태훈 on 2021/05/13.
//  스택 수열 / 스택

func solution() {
    var stack = [Int]()
    var res = [String]()
    var cnt = 1
    var isPossible = true
    
    for _ in 0 ..< Int(readLine()!)! {
        let num = Int(readLine()!)!
        while cnt <= num {
            stack.append(cnt)
            res.append("+")
            cnt += 1
        }
        
        if stack.last == num {
            stack.popLast()
            res.append("-")
        } else {
            isPossible = false
            break
        }
    }
    
    isPossible == false ? print("NO") : res.forEach { print($0) }
}

solution()

File: Baekjoon/Python/BOJ_PS/Stack-Queue/Q1874.py
# 스택 수열 / 스택
import sys
input = sys.stdin.readline

def solution():
    stack = [] 
    res = []
    cnt = 1
    isPossible = True

    for _ in range(int(input())): 
        n = int(input())
        while cnt <= n:
            stack.append(cnt)
            res.append('+')
            cnt += 1
        
        if stack[-1] == n:
            stack.pop()
            res.append('-')
        else:
            isPossible = False
            break

    if isPossible == False:
        print("NO")
    else: 
        for i in res:
            print(i)

solution()